src/main/java/org/springframework/integration/kafka/listener/AbstractOffsetManager.java

```diff
@@ -54,6 +54,8 @@
 
 	protected Map<Partition, Long> initialOffsets;
 
+	protected Map<Partition, Long> highestUpdatedOffsets = new ConcurrentHashMap<Partition, Long>();
```

Comment:
Well, I'd honestly wouldn't want to mix things, and keep the two separate. It could be wasteful, but i'd avoid the interference, to be honest.